python/graphstorm/gconstruct/construct_graph.py

```diff
@@ -370,8 +372,60 @@ def parse_node_data(in_file, feat_ops, node_id_col, label_conf,
             feat_data[key] = val
     return (data[node_id_col], feat_data)
 
+def map_node_ids(src_ids, dst_ids, edge_type, node_id_map, skip_nonexist_edges):
+    """ Map node IDs of source and destination nodes of edges.
+
+    In the ID mapping, we need to handle many errors in the input data. For example,
+    we need to handle the case that endpoint nodes of edges don't exist; we also need
+    to handle the case that the data type of node IDs of the endpoint nodes don't
+    match the data type of the keys of the ID map.
+
+    Parameters
+    ----------
+    src_ids : tensor
+        The source nodes.
+    dst_ids : tensor
+        The destination nodes.
+    edge_type : tuple
+        It contains source node type, relation type, destination node type.
+    node_id_map : dict
+        The key is the node type and value is IdMap or IdentityMap.
+    skip_nonexist_edges : bool
+        Whether or not to skip edges whose endpoint nodes don't exist.
+
+    Returns
+    -------
+    tuple of tensors : the remapped source and destination node IDs.
+    """
+    src_type, _, dst_type = edge_type
+    new_src_ids, idx = node_id_map[src_type](src_ids)
```

Comment:
Does `idx` represent the nodes that exists in node_id_map[src_type] ? Can we give it a better name?

Reply:
node location in `src_ids`